### PR TITLE
fix: Nuxi cjs importModule with jiti

### PR DESCRIPTION
**Pull request by jbmolle**
### 🔗 Linked issue

#1795

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [X] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Solves #1795
We couldn't loadNuxt in CJS because import would return a default key.
By using Jiti we ensure that both in CJS and ESM we get { loadNuxt: Function[] } when calling loadKit.
And we can then destructure const { loadNuxt } = loadKit()

### 📝 Checklist

- [X] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/utils/cjs.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export function requireModule (id: string, paths?: string | string[]) {
 
 export function importModule (id: string, paths?: string | string[]) {
   const resolvedPath = resolveModule(id, paths)
-  return import(pathToFileURL(resolvedPath).href)
+  return jiti(resolvedPath, { interopDefault: true })(resolvedPath)
```
Comment (pi0) [inline packages/nuxi/src/utils/cjs.ts:34]: I'm not sure this is a right fix. `importModule` is intended to use [native ESM import](https://v3.nuxtjs.org/concepts/esm)

**Comment by pi0** [comment]
> We couldn't loadNuxt in CJS because import would return a default key.

`@nuxt/kit` should be imported using `await import('@nuxt/kit')` in the CJS context.

**Comment by jbmolle** [comment]
Yes I understand.
The problem is that `importModule` called in `loadKit` in `framework/packages/nuxi/src/utils/kit.ts` returns `{ loadNuxt, ... }` in ESM and `{ default: { loadNuxt, ... } }` in CJS. So when using @nuxt/kit: latest, we can't start Nuxt.

**Comment by pi0** [comment]
CJS proxy removed from `@nuxt/kit` stub (https://github.com/nuxt/framework/issues/1795#issuecomment-972749713)

**Pull request closed without merging**